instrumentation/p6spy/src/main/java/brave/p6spy/TracingJdbcEventListener.java

```diff
@@ -0,0 +1,88 @@
+package brave.p6spy;
+
+import brave.Span;
+import brave.Tracer;
+import brave.Tracing;
+import com.p6spy.engine.common.StatementInformation;
+import com.p6spy.engine.event.SimpleJdbcEventListener;
+import java.net.InetAddress;
+import java.net.URI;
+import java.sql.Connection;
+import java.sql.SQLException;
+import zipkin.Constants;
+import zipkin.Endpoint;
+import zipkin.TraceKeys;
+
+final class TracingJdbcEventListener extends SimpleJdbcEventListener {
+
+  final String remoteServiceName;
+
+  TracingJdbcEventListener(String remoteServiceName) {
+    this.remoteServiceName = remoteServiceName;
+  }
+
+  @Override public void onBeforeAnyExecute(StatementInformation info) {
+    Tracer tracer = Tracing.currentTracer();
+    if (tracer == null) return;
```

Comment:
This method is special-cased variant of `Tracing.tracer() != null` Anyway the rationale was expanded on here https://github.com/openzipkin/brave/pull/379#discussion_r111049601